Cd relnotes 805xx cds 78504 (#3197)
harness/developer-hub: release-notes/continuous-delivery.md

<Kustomizedep />

## Latest: Version 805xx

### New features and enhancements

This release does not have new features.

### Early access features

This release does not have early access features.

### Fixed issues

> * Fixed a UI issue observed when retrying a pipeline execution: the wrong **Execution Id** was displayed because the UI was evaluating the `sequenceId` based on the root `executionId`. With this fix, the **Execution Id** is based on the back-end data for the current pipeline execution. (CDS-78305)

Reviewer: ```suggestion
* Fixed a UI issue observed where the wrong **Execution Id** was displayed when retrying a pipeline execution. The wrong **Execution Id** was displayed because the UI was evaluating the `sequenceId` based on the root `executionId`. With this fix, the **Execution Id** is based on the backend data for the current pipeline execution. (CDS-78305)
```

> * Fixed the following issue: when a user re-ran a pipeline execution with a different branch selected, the pipeline would execute with the last branch selected. With this fix, the pipeline always runs with the current branch selected in **Run Pipeline**. (CDS-77826)

Reviewer: ```suggestion
* When users reran a remote pipeline with a new branch selected, the pipeline would execute with the previous branch selected. With this fix, the pipeline always runs with the current branch selected in **Run Pipeline**. (CDS-77826)
```

* The following Node.js v20 images are now available on Docker Hub. (CDS-77801)
  - [harnessdev/serverless-preparerollback:3.30.1-1.0.1](https://hub.docker.com/layers/harnessdev/serverless-preparerollback/3.30.1-1.0.1/images/sha256-30994cb1fef9b206babecbe61b10144591db51cddec2e5386a0ab228d542b32f?context=explore)
  - [harnessdev/serverless-deploy:3.30.1-1.0.1](https://hub.docker.com/layers/harnessdev/serverless-deploy/3.30.1-1.0.1/images/sha256-f1ec1e56cb22403e492d8aa9c4f719cd466aea1a06a438d36818ac607e7acff0?context=explore)
  - [harnessdev/serverless-package:3.30.1-1.0.1](https://hub.docker.com/layers/harnessdev/serverless-package/3.30.1-1.0.1/images/sha256-1448b1566186023df78cc79b529c06256e327890b5018ed05632775a7c5c4992?context=explore)
  - [harnessdev/serverless-rollback:3.30.1-1.0.1](https://hub.docker.com/layers/harnessdev/serverless-rollback/3.30.1-1.0.1/images/sha256-c023e84a4a7450c4978535569b03e8909a7e6daa9c2fa0ad4ff70a6ee52e02e7?context=explore)

* Fixed an issue resolving secrets via `<+secrets.getValue("my_secret")>` used in container-based step groups in a Deploy stage.  (CDS-77793, ZD-49391, ZD-49763, ZD-49838, ZD-50014)

> * Fixed an issue when using the OCI Helm connector with anonymous credentials: Harness would incorrectly derive the source port from OCI repository and the connection validation would fail. With this fix, Harness deducts default port 443 if no port is provided in the URL itself. (CDS-77298)

Reviewer: ```suggestion
* When using the OCI Helm connector with anonymous credentials, Harness would incorrectly derive the source port from the OCI repository and the connection validation would fail. With this fix, Harness does not add default port 443 if no port is provided in the URL itself. (CDS-77298)
```

> * Fixed an issue where users could not specify runtime inputs for some advanced settings (such as Looping Strategy) in step groups and step-group templates. With this fix, runtime inputs are supported for all advanced step-group settings. (CDS-77246, ZD-49339, ZD-49427)

Reviewer: ```suggestion
* Fixed an issue where users could not specify runtime inputs for some advanced settings (such as Looping Strategy) in step groups and step group templates. With this fix, runtime inputs are supported for all advanced step group settings. (CDS-77246, ZD-49339, ZD-49427)
```

Reviewer: step group is not hyphenated. it's a compound noun.

* Fixed a UI issue when running a pipeline with an input set that  allows multiple choices (checkboxes) for a field. Previously, if no checkboxes were selected for a field, the pipeline set the value to an empty string. This could cause the pipeline execution to fail. With this fix, the pipeline sets field to undefined if no checkboxes are selected. (CDS-77221)

* Fixed an issue where users could not save modified input sets from from the YAML view. (CDS-77117, ZD-49265)

* Fixed a UI issue observed when retrying a pipeline execution: the wrong **Execution Id** was displayed because the UI was evaluating the `sequenceId` based on the root `executionId`. With this fix, the **Execution Id** is based on the back-end data for the current pipeline execution. (CDS-76697)

* Fixed an issue observed when deploying Artifactory artifacts, where the pipeline execution would save the generic artifact metadata incorrectly. Note that this fix applies to new pipeline executions only. (CDS-76302, ZD-48430)

> * Fixed an issue in the **Run Pipeline** UI when running child pipelines . Previously, **Pipeline Variables** was in a new container, which implied that **Pipeline Variables** was a new stage. (CDS-74138)

Reviewer: ```suggestion
* Fixed an issue in the **Run Pipeline** UI when running chained pipelines. Previously, **Pipeline Variables** was indented in the UI, which implied that **Pipeline Variables** was a new stage. (CDS-74138)
```

Author: @michaelcretzman The SME asked me to use "in a new container" rather than "indented," I'm going to leave this one as-is.

Author: I did change "child pipeline" to "chained pipeline" though

  
  ![](static/cds-74138-00-before.png)

  With this fix, **Pipeline Variables** is indented under the same container to make it clear that these are inputs from the child pipeline. 
  
  ![](static/cds-74138-01-after.png)

  <!-- https://harness.atlassian.net/browse/CDS-74138?focusedCommentId=567699 -->

> * API Call logs now include details such as Response, size, duration, HTTP Verb, and Response code in the summary. (OIP-767)

Reviewer: ```suggestion
* API call logs summary now includes details such as response, size, duration, HTTP verb, and response code. (OIP-767)
```

* Fixed an issue that resulted in Null Pointer Exceptions when running a pipeline manually with a  `<+trigger.connectorRef>` expression. This expression gets its data from the trigger payload. With this fix, the pipeline correctly handles the case where the trigger payload is null.  (CDS-77736, ZD-49685, ZD-49720, ZD-49722)

*  Fixed an intermittent issue where Helm deployment pipelines would report the Helm repository as not found.  (CDS-76919)

<!-- Adding per this comment: https://harness.atlassian.net/browse/CDS-75488?focusedCommentId=567310 -->

* Due to intermittent issues with the `cf` CLI, the Tanzu Application Services (TAS) Rolling deployment step failed to create the application. (CDS-75250)

>   Now, before performing a rolling deployment, the TAS Rolling deployment step first verifies that the application exists. If the application does not exist, it deploys the application without using the rolling deployment strategy. If the application exists, it performs a rolling upgrade.

Reviewer: > it deploys the application without using the rolling deployment strategy

So it uses a Basic strategy?

Author: Yes, I've updated the draft to say Basic strategy
